### PR TITLE
Fix for Patch3D get_facecolor after set_facecolor

### PR DESCRIPTION
**Pull request by watermarkhu**
## PR Summary

Solves #18815 by updating `_facecolor3d` after `set_facecolor`. ~No doc or test as it is small fix.~

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A] New features are documented, with examples if plot related.
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 1]
```diff
@@ -347,6 +347,10 @@ def get_path(self):
     def get_facecolor(self):
         return self._facecolor2d
```
Comment (jklymak) [inline lib/mpl_toolkits/mplot3d/art3d.py:348]: Is this right?

**Review by jklymak — COMMENTED** [review]
This definitely needs a test!

**Comment by tacaswell** [comment]
Looking at this with @QuLogic we think the fix is to remove `obj._facecolor2d` and `obj._facecolor3d` from `Patch3D` and `PathPatch3D` all together and just rely on the behavior of the base for the color.

**Comment by WeatherGod** [comment]
You then lose the ability to do depthshading

**Comment by QuLogic** [comment]
These artists don't do depth shading, AFAICT.

**Comment by WeatherGod** [comment]
you might be right, could I be thinking of Path3DCollections? What is it
that is used by scatter()? There is also the other shading features, but I
think those are also done to Path3DCollections.

**Comment by QuLogic** [comment]
Replaced by #19448.

**Comment by tacaswell** [comment]
@watermarkhu Thank you for your work on this.  Even though we did not fix it the way you proposed, we appreciate you reporting the issue and proposing a fix! 

Hopefully we will hear from you again.

**Pull request closed without merging**